common/src/main/java/com/wynntils/handlers/item/ItemHandler.java

```diff
@@ -88,20 +89,17 @@ private void onItemStackUpdate(ItemStack existingItem, ItemStack newItem) {
             return;
         }
 
-        // Check if item type, damage and count matches, if not, it's definitely a new item
```

Comment:
This is a very handy shortcut. Why did you remove this? Did any of item type, damage or count change when applying a skin?

Reply:
The item has to fundamentally change for a skin to work, so yes. I'm unsure if there's cases where the item itself will change or if it's only the item damage, but I'd say it's safest to remove the check altogether for this context

Reply:
weapon skins change the model of the weapon; that can't be achieved without at the very least changing the item's damage value

Reply:
Ok, I have checked a bit. As far as I could tell, *all* weapon skins are `diamond_shovel`. And head-wear skins are `diamond_pickaxe`. That means we could accept that these two items are "wildcard" items that match anything. So in similarStack, the short-circuit logic will return false if the two Items are different, but not if one of the two is a "wildcard" item.

something like this:
```
    private boolean similarStack(ItemStack firstItem, ItemStack secondItem) {
        if (!firstItem.getItem().equals(secondItem.getItem()) &&!isWildcardItem(firstItem) && !isWildcardItem(secondItem)) return false;
       
```
and `isWildcardItem` checks for equality with `Items.DIAMOND_PICKAXE` or `Items.DIAMOND_SHOVEL`.

Reply:
I realize this hard-codes a bit Wynncraft-y behavior, but I'm willing to sacrifice that to keep the high-performing short-circuit logic of `similarStack`.